File: app/lib/src/models/resource.dart
class Resource {
  final String uuid;
  final String title;
  final String content;
  final String? imageUrl;
  final String? fileUrl;
  final List<Map<String, dynamic>> categories;
  final bool isPublished;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Resource({
    required this.uuid,
    required this.title,
    required this.content,
    this.imageUrl,
    this.fileUrl,
    this.categories = const [],
    this.isPublished = true,
    this.createdAt,
    this.updatedAt,
  });

  factory Resource.fromJson(Map<String, dynamic> json) {
    return Resource(
      uuid: json['uuid'],
      title: json['title'],
      content: json['content'],
      imageUrl: json['image_path'] ?? json['image_url'],
      fileUrl: json['file_url'],
      categories: List<Map<String, dynamic>>.from(json['categories'] ?? []),
      isPublished: json['is_published'] ?? true,
      createdAt: json['created_at'] != null ? DateTime.parse(json['created_at']) : null,
      updatedAt: json['updated_at'] != null ? DateTime.parse(json['updated_at']) : null,
    );
  }
} 

File: backend/app/Models/Resource.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Support\Facades\Storage;

class Resource extends Model
{
    use HasFactory, SoftDeletes, HasUuids;

    protected $fillable = [
        'title',
        'content',
        'image_url',
        'is_published',
        'created_by'
    ];

    protected $casts = [
        'is_published' => 'boolean',
    ];

    protected $appends = ['image_path'];

    /**
     * Get the columns that should receive a unique identifier.
     *
     * @return array
     */
    public function uniqueIds(): array
    {
        return ['uuid'];
    }

    public function getImagePathAttribute()
    {
        return $this->image_url ? Storage::disk('public')->url($this->image_url) : null;
    }

    /**
     * Get the route key for the model.
     */
    public function getRouteKeyName()
    {
        return 'uuid';
    }

    public function creator()
    {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function categories()
    {
        return $this->belongsToMany(Category::class, 'resource_categories');
    }
} 